src/test/groovy/net/vivin/gradle/versioning/VersionUtilsTests.groovy

```diff
@@ -0,0 +1,40 @@
+package net.vivin.gradle.versioning
+
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.internal.storage.file.FileRepository
+import org.eclipse.jgit.lib.Repository
+import org.gradle.testfixtures.ProjectBuilder
+import org.testng.annotations.BeforeMethod
+import org.testng.annotations.Test
+
+import static org.testng.Assert.assertFalse
+
+class VersionUtilsTests extends TestNGRepositoryTestCase {
+
+    @Override
+    @BeforeMethod
+    void setUp() {
+        super.setUp()
+        Repository bare = createBareRepository()
+        File clone = createTempDirectory("clone")
+        Git git = Git.cloneRepository().setURI(bare.getDirectory().getAbsolutePath()).setDirectory(clone).call()
+        db = (FileRepository) git.repository
+
+        testRepository = new TestRepository(db)
+
+        project = ProjectBuilder.builder()
+                .withProjectDir(testRepository.repository.getDirectory().getParentFile())
+                .build()
+        project.apply plugin: 'net.vivin.gradle-semantic-build-versioning'
+    }
+
+    @Test
+    void testTaggedVersionIsRecognizedAsNonSnapshot() {
+        testRepository.commitAndTag("0.0.1")
+
+        SemanticBuildVersion version = (SemanticBuildVersion) project.getVersion()
+
+        version.versionUtils.determineVersion()
+        assertFalse(version.snapshot, "Tagged version is snapshot version")
```

Comment:
Minor: I usually write assertions using "should" or "should not" so that it's clear what is expected. Maybe reword this to "Tagged version should not be snapshot version"?